auth/impersonate/example_test.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package impersonate_test
+
+import (
+	"log"
+
+	"cloud.google.com/go/auth/httptransport"
+	"cloud.google.com/go/auth/impersonate"
+)
+
+func ExampleNewCredentialTokenProvider_serviceAccount() {
```

Comment:
Nit: Upper case after underscore.

Reply:
This is actually a go convetion to have multiple named examples in godoc. Example: https://pkg.go.dev/google.golang.org/api/impersonate#pkg-examples

Reply:
Ah, you're correct, sorry! 

> Multiple examples can be provided for a given identifier by using a suffix beginning with an underscore followed by a lowercase letter. 

-- https://go.dev/blog/examples